consecutive-numbers/tealsGetConsecutiveNumbers.js

```diff
@@ -0,0 +1,44 @@
+//O(N) time O(1) space.
+//works if array has negative integers as well
+
+getConsecutiveNums = (arr) => {
+  if (!arr[0]) return '';
+  if (arr.length === 1) return arr[0].toString();
+
+  let result = '';
+  let i = 0;
+
+  while (i<arr.length){
+    let start = arr[i];
+    let j = i + 1;
+    let end = arr[j];
+
+    while (end <=(arr[j-1] + 1) && i<arr.length && j<arr.length){
+      j++;
+      end = arr[j];
+    } 
+
+    if (arr[i] == arr[j-1]){
+      result += start.toString() + ', ';
+      i = j;
+    } else {
+      result += start.toString() + '-' + arr[j-1].toString() + ', '
+      i = j;
```

Comment:
see if you can use the `.toString()` method a little higher up here. maybe at the initialization steps?

Reply:
Thanks.  It actually didn't need it at all... dynamic typing drives me nuts sometimes, but is helpful here!